### PR TITLE
STR-722: refactor random generation to `CryptoRngCore`

### PR DESCRIPTION
**Pull request by AaronFeickert**
## Description

This PR refactors the trait bound `CryptoRng + RngCore` to `CryptoRngCore` from `rand_core` and performs associated import cleanup.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature/Enhancement (non-breaking change which adds functionality or enhances an existing one)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Refactor
- [ ] New or updated tests
- [x] Dependency Update

## Notes to Reviewers

None.

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code where necessary.
- [x] I have updated the documentation if needed.
- [x] My changes do not introduce new warnings.
- [x] I have added tests that prove my changes are effective or that my feature works.
- [x] New and existing tests pass with my changes.

## Related Issues

None.

### CODE REVIEW

**Comment by storopoli** [comment]
The idea is to break up STR-722 into small easy-to-review PRs?
Then in the final one remove `rand` altogether for `rand-core`?

**Comment by AaronFeickert** [comment]
> The idea is to break up STR-722 into small easy-to-review PRs? Then in the final one remove `rand` altogether for `rand_core`?

The primary goal is to use `CryptoRngCore` instead of `CryptoRng + RngCore` to reduce the likelihood of missing `CryptoRng`. However, `CryptoRngCore` is only provided by `rand_core`, whereas `CryptoRng` and `RngCore` are _also_ provided by `rand`. With a few small extra import updates, it happened to be possible to swap out `rand` for `rand_core` as a dependency in a couple of crates, so I went ahead and did that to clean things up a bit.

There are a few places where we really do need `rand` to get access to sampling and shuffling, so it will stick around. Initially, I went a bit wild with swapping out to `rand_core`, but realized that's out of scope here and not the intent of the issue.

File: bin/strata-cli/src/cmd/faucet.rs [hunk 1]
```diff
@@ -71,15 +70,17 @@ pub async fn faucet(args: FaucetArgs, seed: Seed, settings: Settings) {
...
         prehash.clone(),
         challenge.difficulty,
         solution.to_le_bytes(),
     ) {
         solution += 1;
-        if (0..100).sample_single(&mut OsRng) == 0 {
+        if counter % 100 == 0 {
             pb.set_message(format!("Trying {solution}"));
         }
+        counter += 1;
```
Comment (storopoli) [inline bin/strata-cli/src/cmd/faucet.rs:83]: This is not calling `sample_single` anymore.
Wasn't this needed for the PoW?

**Review by storopoli — CHANGES_REQUESTED** [review]
Just one question to clarify otherwise good to go.

**Review by storopoli — APPROVED** [review]
ACK 4605cf31980e1dd6934daa3a371cbaa1da226d9f

**Review by delbonis — APPROVED** [review]

**Pull request merged**